AdminPortal/src/pages/dashboard.tsx

```diff
@@ -1,6 +1,284 @@
+
+      let newState: boolean[] = [];
+
+      for (let i = 0; i < numTabs; i++) {
+        if (i === indexToFlip) {
+          newState.push(!prevState[i]);
+        } else {
+          newState.push(prevState[i]);
+        }
+      }
+
+      return newState;
+    });
+  };
+
+  useEffect(() => {
+    setItems(
+      items.map((item) =>
+        item.category === activeTab
+          ? {
+              ...item,
+              isChecked: selectAllClicked[tabMapping.get(activeTab) - 1],
+            }
+          : item
+      )
+    );
+  }, [selectAllClicked]);
+
+  const handleItemCheckbox = (id: number): void => {
+    setItems(
+      items.map((item) =>
+        item.id === id ? { ...item, isChecked: !item.isChecked } : item
+      )
+    );
+  };
+
+  return (
+    <div className={styles.page}>
+      <Sidebar currentPage="Dashboard"></Sidebar>
+      <div className={styles.outer}>
+        <h1 className={styles.title}>Driver Registration</h1>
+
+        <div className={styles.tabs}>
+          <button
+            className={
+              activeTab !== 'Needs Review'
+                ? styles.tabsButton
+                : styles.tabsButtonSelected
+            }
+            onClick={() => handleTabClick('Needs Review')}
+          >
+            Needs Review
+          </button>
+          <button
+            className={
+              activeTab !== 'In Review'
+                ? styles.tabsButton
+                : styles.tabsButtonSelected
+            }
+            onClick={() => handleTabClick('In Review')}
+          >
+            In Review
+          </button>
+          <button
+            className={
+              activeTab !== 'Verified'
+                ? styles.tabsButton
+                : styles.tabsButtonSelected
+            }
+            onClick={() => handleTabClick('Verified')}
+          >
+            Verified
+          </button>
+          <button
+            className={
+              activeTab !== 'Disapproved'
+                ? styles.tabsButton
+                : styles.tabsButtonSelected
+            }
+            onClick={() => handleTabClick('Disapproved')}
+          >
+            Disapproved
+          </button>
+        </div>
+
+        <hr className={styles.horizontalLine}></hr>
+
+        <div className={styles.dropdownAndExportBar}>
+          <Select
+            options={options}
+            onChange={handleDropdownClick}
+            styles={customStyle}
+            placeholder="Change Status To"
+            blurInputOnSelect={true}
+          />
+
+          <button className={styles.exportButton}>Export to CSV File</button>
+        </div>
+
+        <table>
+          <thead className={styles.tableHead}>
+            <tr>
+              <th className={styles.tableHeadItem}>
+                <input
+                  type="checkbox"
+                  className={styles.checkbox}
+                  checked={selectAllClicked[tabMapping.get(activeTab) - 1]}
```

Comment:
What's the point of -1 here? It's not very obvious in the code what's going on here

Reply:
There's a simpler way to approach selectAll here: just have selectAllChecked be a boolean, and have checking it setSelectAllChecked to true and check all visible boxes. Then, when we uncheck a box or move tabs, we setSelectAllChecked to false

Reply:
At the moment, it is a vector that keeps a track of the which tabs out of the 4 have the selectAll option enabled/disabled, so not sure how changing that to a boolean simply would work as suggested in the comment. I understand that this is not apparent looking at the the naming of the react state. Would it be fine if I just add comments explaining that?

Reply:
As discussed in the meeting, I have made the change you suggested. Now when a user switches tabs, selected rows are automatically unchecked and the selectAll variable is a simple boolean